test/e2e/redundancy/leader_election.go

```diff
@@ -0,0 +1,135 @@
+			}
+
+			clusterName := framework.TestContext.ClusterIDs[cluster]
+
+			gatewayPod := f.AwaitActiveGatewayPod(cluster, nil)
+			Expect(gatewayPod).ToNot(BeNil(), "Did not find an active gateway pod")
+
+			restartCount := gatewayPod.Status.ContainerStatuses[0].RestartCount
+
+			gateway := f.AwaitGatewayWithStatus(cluster, gatewayPod.Spec.NodeName, subv1.HAStatusActive)
+
+			By(fmt.Sprintf("Found submariner Gateway %q on cluster %q", gateway.Name, clusterName))
+
+			By("Updating submariner-gateway Role to remove Lease update permission")
+
+			updateLeaseUpdatePermission(cluster, gateway.Namespace, true)
+
+			DeferCleanup(func() {
+				updateLeaseUpdatePermission(cluster, gateway.Namespace, false)
+			})
+
+			By(fmt.Sprintf("Ensure Gateway %q is updated to passive", gateway.Name))
+
+			f.AwaitGatewaysWithStatus(cluster, subv1.HAStatusPassive)
+
+			tcp.RunConnectivityTest(tcp.ConnectivityTestParams{
+				Framework:             f.Framework,
+				ToEndpointType:        tcp.PodIP,
+				Networking:            framework.PodNetworking,
+				FromCluster:           framework.ClusterA,
+				FromClusterScheduling: framework.NonGatewayNode,
+				ToCluster:             framework.ClusterB,
+				ToClusterScheduling:   framework.NonGatewayNode,
+			})
+
+			By("Updating submariner-gateway Role to add Lease update permission")
+
+			updateLeaseUpdatePermission(cluster, gateway.Namespace, false)
+
+			By(fmt.Sprintf("Ensure Gateway %q is updated to active", gateway.Name))
+
+			f.AwaitGatewayWithStatus(cluster, gatewayPod.Spec.NodeName, subv1.HAStatusActive)
+
+			tcp.RunConnectivityTest(tcp.ConnectivityTestParams{
+				Framework:             f.Framework,
+				ToEndpointType:        tcp.PodIP,
+				Networking:            framework.PodNetworking,
+				FromCluster:           framework.ClusterA,
+				FromClusterScheduling: framework.GatewayNode,
+				ToCluster:             framework.ClusterB,
+				ToClusterScheduling:   framework.GatewayNode,
+			})
+
+			gatewayPod, err := framework.KubeClients[cluster].CoreV1().Pods(gatewayPod.Namespace).Get(context.Background(),
+				gatewayPod.Name, metav1.GetOptions{})
+			Expect(err).To(Succeed())
+
+			Expect(gatewayPod.Status.ContainerStatuses[0].RestartCount).To(Equal(restartCount),
+				"Gateway pod %q was restarted", gatewayPod.Name)
+		})
+	})
+})
+
+func updateLeaseUpdatePermission(cluster framework.ClusterIndex, ns string, remove bool) {
+	err := util.Update[*rbacv1.Role](context.Background(), resource.ForRole(framework.KubeClients[cluster], ns),
+		&rbacv1.Role{
+			ObjectMeta: metav1.ObjectMeta{Name: "submariner-gateway"},
+		},
+		func(role *rbacv1.Role) (*rbacv1.Role, error) {
+			for i := range role.Rules {
+				if role.Rules[i].APIGroups[0] == coordinationv1.GroupName && role.Rules[i].Resources[0] == "leases" {
+					if remove {
+						role.Rules[i].Verbs, _ = slices.Remove(role.Rules[i].Verbs, "update", func(e string) string {
```

Comment:
How about passing in `slices.Remove` or `slices.AppendIfNotPresent` instead of a boolean? Both functions have the same signature.

Reply:
yeah I could try that.